src/core/base-runner.js

```diff
@@ -27,11 +27,13 @@ function toRunnable(plug) {
         performance.mark(name + "-start");
       }
       try {
-        // Modern plugins are async or normal functions, take zero or one argument (conf)
         if (plug.run.length <= 1) {
           await plug.run(config);
           resolve();
         } else {
+          console.warn(
+            `Plugin ${name} uses a deprecated callback signature. Return a Promise instead.`
```

Comment:
We should probably link to: https://github.com/w3c/respec/wiki/Developers-Guide#plugins

Reply:
Should we use `pub("warn")` for this?

Reply:
hmm... It's really a developer error, not an Editor error.

Reply:
(so, I say we just stick with console.warn)